internal/core/textlayout.rs

```diff
@@ -252,6 +252,39 @@ impl<'a, Font: AbstractFont> TextParagraphLayout<'a, Font> {
             Err(position) => position,
         }
     }
+
+    /// Returns the bytes offset for the given position
+    pub fn byte_offset_for_position(
+        &self,
+        pos: (Font::Length, Font::Length),
```

Comment:
I think you could probably make the code a little easier to read (pos_x instead of pos.0) if you destructure the tuple:

```suggestion
        (pos_x, pos_y): (Font::Length, Font::Length),
```